lib/sanbase/social_data/social_volume.ex

```diff
@@ -62,78 +65,60 @@ defmodule Sanbase.SocialData.SocialVolume do
     end
   end
 
-  def topic_search(search_text, from, to, interval, source)
-      when source in [:all, "all", :total, "total"] do
-    result =
-      @sources
-      |> Sanbase.Parallel.flat_map(
-        fn source ->
-          {:ok, result} = topic_search(search_text, from, to, interval, source)
-          result
-        end,
-        max_concurrency: 4
-      )
-      |> Sanbase.Utils.Transform.sum_by_datetime(:mentions_count)
-
-    {:ok, result}
-  end
-
-  def topic_search(search_text, from, to, interval, source) do
-    topic_search_request(search_text, from, to, interval, source)
+  defp social_volume_selector_handler(%{slug: slug}) do
+    slug
+    |> Project.by_slug(only_preload: [:social_volume_query])
     |> case do
-      {:ok, %HTTPoison.Response{status_code: 200, body: body}} ->
-        {:ok, result} = Jason.decode(body)
-        topic_search_result(result)
+      %Project{social_volume_query: %Sanbase.Model.Project.SocialVolumeQuery{query: query_text}}
```

Comment:
```suggestion
      %Project{social_volume_query: %{query: query_text}}
```
structs also match against maps, you can do that to be short - also you don't expect any other struct here so it should be ok

Reply:
If you want to keep it, you have it aliased so just `%SocialVolumeQuery` would do